src/Folder.res

```diff
@@ -48,18 +48,12 @@ let addFolderToRoot = (rootFolder: t, targetFolderId: int, name: string, id: int
   }
 }
 
+type componentProps = {currentFolder: t, handleClick: int => unit, recLevel: int}
```

Comment:
It's a bit ugly that the props reside outside of the component, but otherwise you would need to write it out both for module type and module.